core/src/main/java/site/ycsb/TerminatorThread.java

```diff
@@ -28,26 +28,29 @@
 public class TerminatorThread extends Thread {
 
   private final Collection<? extends Thread> threads;
-  private long maxExecutionTime;
-  private Workload workload;
-  private long waitTimeOutInMS;
+  private final long maxExecutionTime;
+  private final Workload workload;
+  private final long waitTimeOutInMS;
 
-  public TerminatorThread(long maxExecutionTime, Collection<? extends Thread> threads,
-                          Workload workload) {
+  public TerminatorThread(long maxExecutionTime, Collection<? extends Thread> threads, Workload workload) {
     this.maxExecutionTime = maxExecutionTime;
     this.threads = threads;
     this.workload = workload;
-    waitTimeOutInMS = 2000;
-    System.err.println("Maximum execution time specified as: " + maxExecutionTime + " secs");
+    this.waitTimeOutInMS = 2000;
   }
 
   public void run() {
+    System.err.println("Maximum execution time specified as: " + maxExecutionTime + " secs");
     try {
       Thread.sleep(maxExecutionTime * 1000);
     } catch (InterruptedException e) {
       System.err.println("Could not wait until max specified time, TerminatorThread interrupted.");
       return;
     }
+    stopWorkload();
+  }
+
+  public void stopWorkload() {
     System.err.println("Maximum time elapsed. Requesting stop for the workload.");
```

Comment:
This message should be in the `run()` method.

Reply:
Fixed